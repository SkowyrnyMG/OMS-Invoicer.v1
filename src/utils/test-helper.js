import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
