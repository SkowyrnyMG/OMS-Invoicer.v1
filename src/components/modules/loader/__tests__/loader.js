import React from 'react';

import { snapShotTest } from 'utils/tests/test-helper';
import Loader from '../loader';

describe('Loader', () => {
  test('should display in the document with correct styels', () => {
    snapShotTest(<Loader />);
  });
});
