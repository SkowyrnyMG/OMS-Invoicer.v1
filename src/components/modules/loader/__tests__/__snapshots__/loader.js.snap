// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should display in the document with correct styels 1`] = `
.c0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  width: 100%;
  z-index: 50000;
  -webkit-transition: 0.25 all;
  transition: 0.25 all;
}

.c1 {
  position: relative;
  width: 8rem;
  height: 8rem;
  border-radius: 50%;
}

.c2 {
  position: absolute;
  width: 1.2rem;
  height: 1.2rem;
  background: #446DF6;
  border-radius: 50%;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

.c2:first-of-type {
  top: 0%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c2:nth-of-type(2) {
  top: 50%;
  right: 0%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c2:nth-of-type(3) {
  bottom: 0%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c2:nth-of-type(4) {
  top: 50%;
  left: 0%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c3 {
  margin-top: 2rem;
  font-weight: 700;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
    <div
      class="c2"
    />
    <div
      class="c2"
    />
    <div
      class="c2"
    />
  </div>
  <span
    class="c3"
  >
    Loading...
  </span>
</div>
`;
